cmake/cores/Cortex-M7F.cmake

```diff
@@ -28,7 +28,9 @@ set(CMAKE_CXX_LINK_FLAGS "${CMAKE_CXX_LINK_FLAGS} \
 elseif(MBED_TOOLCHAIN STREQUAL "ARM")
 
 set(ARM_FLAGS " \
-    --cpu=Cortex-M7.fp.sp \
+    -mcpu=cortex-m7 \
+    -mfpu=fpv5-sp-d16 \
+    -mfloat-abi=hard \
```

Comment:
Why, for the same target, should GCC be using soft float but ARM use hard float?

Reply:
Maybe we need to switch to hardfloat for GCC (separate commit)

Reply:
As discussed internally, this is still an issue on master as weel, Kevin looked closely at the problem some time ago: https://github.com/ARMmbed/mbed-os/issues/9153